File: src/main/java/fr/lewon/web/bot/methods/FightChampionMethod.java
package fr.lewon.web.bot.methods;

import fr.lewon.bot.errors.BotRunnerException;
import fr.lewon.bot.props.BotPropertyDescriptor;
import fr.lewon.bot.props.BotPropertyStore;
import fr.lewon.bot.props.BotPropertyType;
import fr.lewon.bot.runner.BotRunner;
import fr.lewon.web.bot.HHBot;
import fr.lewon.web.bot.entities.champions.ChampionPremise;
import fr.lewon.web.bot.operations.ChampionFightOperation;
import fr.lewon.web.bot.util.HHRequestProcessor;
import fr.lewon.web.bot.util.HHSession;
import fr.lewon.web.bot.util.HHSessionManager;
import fr.lewon.web.bot.util.HtmlAnalyzer;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class FightChampionMethod extends HHBotProcessor {

    private static final String CHAMPION_ID_KEY = "CHAMPION_ID";

    public FightChampionMethod(String id, String label) {
        super(id, label, Arrays.asList(new BotPropertyDescriptor[]{new BotPropertyDescriptor("CHAMPION_ID", BotPropertyType.INTEGER, null, "Id of the champion", true, false)}));
    }


    @Override
    protected Object doProcess(BotRunner<HHBot> runner, HHSessionManager sessionManager, HHRequestProcessor requestProcessor, BotPropertyStore params) throws BotRunnerException {
        int id = ((Integer) params.get("CHAMPION_ID")).intValue();

        try {
            HHSession session = (HHSession) sessionManager.getSession(requestProcessor);
            String championsContent = requestProcessor.getChampionsMapContent(session);
            List<ChampionPremise> championPremises = HtmlAnalyzer.INSTANCE.getChampionsIds(championsContent);


            ChampionPremise championPremise = championPremises.stream()
                    .filter(c -> (id == c.getChampionId().intValue()))
                    .findFirst()
                    .orElse(null);

            if (championPremise == null) {
                return "Invalid id, no champion found";
            }
        } catch (Exception e) {
            return "ERROR : " + e.getMessage();
        }

        runner.addAction(new ChampionFightOperation(id));
        return "Now fighting champion " + id;
    }


    @Override
    protected List<BotPropertyDescriptor> getSpecificParamBuilders(BotRunner<HHBot> runner) {
        return new ArrayList();
    }
}

File: src/main/kotlin/fr/lewon/bot/hh/operations/FightChampionMethod.kt
package fr.lewon.bot.hh.operations

import fr.lewon.bot.runner.Bot
import fr.lewon.bot.runner.bot.operation.BotOperation
import fr.lewon.bot.runner.bot.operation.OperationResult
import fr.lewon.bot.runner.bot.props.BotPropertyDescriptor
import fr.lewon.bot.runner.bot.props.BotPropertyStore
import fr.lewon.bot.runner.bot.props.BotPropertyType

class FightChampionMethod : BotOperation("Fight champion") {

    companion object {
        private const val CHAMPION_ID_PARAM = "CHAMPION_ID"
    }

    override fun getNeededProperties(bot: Bot): List<BotPropertyDescriptor> {
        return listOf(BotPropertyDescriptor(CHAMPION_ID_PARAM, BotPropertyType.INTEGER, null, "Id of the champion", isNeeded = true, isNullable = false))
    }

    override fun run(bot: Bot, paramsPropertyStore: BotPropertyStore): OperationResult {
//        val id = paramsPropertyStore.getByKey(CHAMPION_ID_PARAM) as Int
//        try {
//            val session = bot.sessionManager.getSession() as HHSession
//            val championsContent = HHRequestProcessor().getChampionsMapContent(session)
//            val championPremises = HtmlAnalyzer.INSTANCE.getChampionsIds(championsContent)
//            val championPremise = championPremises!!.stream()
//                    .filter { c: ChampionPremise? -> id == c.getChampionId().toInt() }
//                    .findFirst()
//                    .orElse(null)
//                    ?: return "Invalid id, no champion found"
//        } catch (e: Exception) {
//            return "ERROR : " + e.message
//        }ab
//        runner.addAction(ChampionFightOperation(id))
//        return "Now fighting champion $id"
        return OperationResult(false, null, null)
    }
}